chcemvediet/apps/anonymization/tests/test_management_commands.py

```diff
@@ -0,0 +1,114 @@
+        call_command(u'attachment_anonymization', self.attachment.pk)
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.file.read(), u'Content from stdin.')
+
+    def test_file_argument_and_stdin_together_may_not_be_omitted(self):
+        self.addCleanup(setattr, sys, u'stdin', sys.stdin)
+        sys.stdin = StringIO(u'')
+        with self.assertRaisesMessage(CommandError, u'No content given.'):
+            call_command(u'attachment_anonymization', self.attachment.pk)
+
+    def test_preferred_content_source_is_file(self):
+        self.addCleanup(setattr, sys, u'stdin', sys.stdin)
+        sys.stdin = StringIO(u'Content from stdin.')
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename)
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.file.read(), u'Default testing content')
+
+    def test_file_is_invalid(self):
+        filename = u'/tmp/invalid.txt'
+        with self.assertRaisesMessage(CommandError, u'Could not open file: '.format(filename)):
+            call_command(u'attachment_anonymization', self.attachment.pk, filename)
+
+    def test_content_type_option(self):
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename, content_type=u'application/pdf')
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.content_type, u'application/pdf')
+
+    def test_content_type_option_default_value_if_omitted(self):
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename)
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.content_type, u'text/plain')
+
+    def test_debug_option(self):
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename, debug=u'debug')
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.debug, u'debug')
+
+    def test_debug_option_default_value_if_omitted(self):
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename)
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.debug, u'')
+
+    def test_force_option(self):
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename)
+        attachment_finalization1 = AttachmentFinalization.objects.get(attachment=self.attachment)
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename, force=True)
+        attachment_finalization2 = AttachmentFinalization.objects.get(attachment=self.attachment)
```

Comment:
@viliambalaz By sa zislo asi assertnut, ze `attachment_finalization2` je ine nez `attachment_finalization1`

Reply:
:ok: dakujem. Nad tym som tiez uvazoval, ale rozhodol som sa to nekontrolovat. Pretoze najprv sa vymaze stara `attachment_finalization1` az tak sa vytvori nova. Teoreticky je mozne ze `attachment_finalization1.pk` sa prideli `attachment_finalization2.pk`. V nasom pripade to vsak nenastane, kedze `pk` je autoincrement.